packages/next/next-server/server/next-server.ts

```diff
@@ -520,10 +520,22 @@ export default class Server {
     )
   }
 
-  private __sendPayload(res: ServerResponse, payload: any, header: string) {
+  private __sendPayload(
+    res: ServerResponse,
+    payload: any,
+    type: string,
+    revalidate?: number | false
+  ) {
     // TODO: ETag? Cache-Control headers? Next-specific headers?
-    res.setHeader('Content-Type', header)
+    res.setHeader('Content-Type', type)
     res.setHeader('Content-Length', Buffer.byteLength(payload))
+
+    if (revalidate) {
+      res.setHeader(
+        'Cache-Control',
+        `s-maxage=${revalidate}, stale-while-revalidate`
```

Comment:
```suggestion
        `s-maxage=${revalidate}, stale-while-revalidate, stale-if-error`
```

Reply:
Should we do a little validation in `revalidate` to make sure the number is sane?

Reply:
Added